frontend/taipy-gui/src/context/patch.spec.ts

```diff
@@ -0,0 +1,91 @@
+        });
+
+        it("should not create nested objects if they don't exist", () => {
+            const obj = {};
+            const originalObj = JSON.stringify(obj);
+            const newObj = patchValue(obj, { a: { b: { c: 1 } } });
+            expect(JSON.stringify(newObj)).toBe(originalObj);
+        });
+
+        it("should handle array indices", () => {
+            const obj = { a: { b: [1, 2, 3] } };
+            const newObj = patchValue(obj, { a: { b: { 1: 4 } } });
+            expect(newObj.a.b[1]).toBe(4);
+        });
+        it("should replace array", () => {
+            const obj = { a: { b: [1, 2, 3] } };
+            const newObj = patchValue(obj, { a: { b: [0] as unknown as number } });
+            expect(newObj.a.b.length).toBe(1);
+            expect(newObj.a.b[0]).toBe(0);
+        });
+        it("should update array", () => {
+            const obj = { a: { b: [1, 2, 3] } };
+            const newObj = patchValue(obj, { a: { b: { 2: [-1, -2] as unknown as number } } });
+            expect(newObj.a.b.length).toBe(4);
+            expect(newObj.a.b[0]).toBe(1);
+            expect(newObj.a.b[1]).toBe(2);
+            expect(newObj.a.b[2]).toBe(-1);
+            expect(newObj.a.b[3]).toBe(-2);
+        });
+        it("should not create nested arrays if they don't exist", () => {
+            const obj = {};
+            const originalObj = JSON.stringify(obj);
+            const newObj = patchValue(obj, { a: { b: { 0: 1 } } });
+            expect(JSON.stringify(newObj)).toBe(originalObj);
+        });
+
+        it("should not modify the original object if the value is the same", () => {
+            const obj = { a: { b: { c: 1 } } };
+            const originalObj = JSON.stringify(obj);
+            const newObj = patchValue(obj, { a: { b: { c: 1 } } });
+            expect(JSON.stringify(newObj)).toBe(originalObj);
+        });
+    });
+    describe("remove", () => {
+        it("should remove the value at the specified path", () => {
+            const obj = { a: { b: { c: 1, d: 2 } } };
+            const newObj = patchValue(obj, undefined, { a: { b: { c: null } } });
+            expect(newObj.a.b.c).toBeUndefined();
+            expect(newObj.a.b.d).toBe(2);
+        });
+        it("should handle array indices", () => {
+            const obj = { a: { b: [1, 2, 3] } };
+            const newObj = patchValue(obj, undefined, { a: { b: { 1: null } } });
+            expect(newObj.a.b.length).toBe(2);
+            expect(newObj.a.b[0]).toBe(1);
+            expect(newObj.a.b[1]).toBe(3);
+        });
+        it("should not modify the original object if the path does not exist", () => {
+            const obj = { a: { b: { c: 1 } } };
+            const newObj = patchValue(obj, undefined, { a: { b: { d: null } } });
+            expect(newObj).toStrictEqual(obj);
+        });
+        it("should not modify the original object if the value to remove is not null", () => {
```

Comment:
Do you think this use case is really important?
I think it makes sense, but is also a constraint that I don't find that helpful...

Reply:
debatable ... it is easier technically to test for null but we could test for not object or null ...

Reply:
To me, a property is there, no matter what the value is, should be dropped.